app/js/FileHandler.js

```diff
@@ -1,235 +1,202 @@
+      // documentation below, we have already made a 'HEAD' request in
+      // checkIfFileExists so we only have "eventual consistency" if we try
+      // to stream it from S3 here.  This was a cause of many 403 errors.
+      //
+      // "Amazon S3 provides read-after-write consistency for PUTS of new
+      // objects in your S3 bucket in all regions with one caveat. The
+      // caveat is that if you make a HEAD or GET request to the key name
+      // (to find if the object exists) before creating the object, Amazon
+      // S3 provides eventual consistency for read-after-write.""
+      // https://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel
+      const readStream = fs.createReadStream(convertedFsPath)
+      readStream.on('end', function() {
+        LocalFileWriter.deleteFile(convertedFsPath, function() {})
+      })
+      callback(null, readStream)
+    }
+  )
+}
 
-      logger.log({ opts }, 'converting file depending on opts')
+function _convertFile(bucket, originalKey, opts, callback) {
+  _writeFileToDisk(bucket, originalKey, opts, function(err, originalFsPath) {
+    if (err) {
+      return callback(
+        new ConversionError('unable to write file to disk').withCause(err)
+      )
+    }
 
-      if (opts.format != null) {
-        return FileConverter.convert(originalFsPath, opts.format, done)
-      } else if (opts.style === 'thumbnail') {
-        return FileConverter.thumbnail(originalFsPath, done)
-      } else if (opts.style === 'preview') {
-        return FileConverter.preview(originalFsPath, done)
-      } else {
+    const done = function(err, destPath) {
+      if (err) {
+        logger.err(
+          { err, bucket, originalKey, opts: _scrubSecrets(opts) },
+          'error converting file'
+        )
         return callback(
-          new Error(
-            `should have specified opts to convert file with ${JSON.stringify(
-              opts
-            )}`
-          )
+          new ConversionError('error converting file').withCause(err)
         )
       }
-    })
-  },
-
-  _writeS3FileToDisk(bucket, key, opts, callback) {
-    return PersistorManager.getFileStream(bucket, key, opts, function(
-      err,
-      fileStream
-    ) {
-      if (err != null) {
-        return callback(err)
-      }
-      return LocalFileWriter.writeStream(fileStream, key, callback)
-    })
-  },
-
-  getDirectorySize(bucket, project_id, callback) {
-    logger.log({ bucket, project_id }, 'getting project size')
-    return PersistorManager.directorySize(bucket, project_id, function(
-      err,
-      size
-    ) {
-      if (err != null) {
-        logger.err({ bucket, project_id }, 'error getting size')
-      }
-      return callback(err, size)
-    })
-  },
-
-  _scrubSecrets(opts) {
-    const safe = Object.assign({}, opts)
-    delete safe.credentials
-    return safe
-  }
+      LocalFileWriter.deleteFile(originalFsPath, function() {})
+      callback(err, destPath)
```

Comment:
It's a parameter from the callback passed to `_writeFileToDisk`.

Reply:
Right, but previously it was passed to the `callback` here. Was it intentionally removed?

Reply:
Ah, I understand. :-) - Yes, we don't need it. It was just used to clean up the temporary file, but we're doing this locally instead because it's less messy.